library/billing/src/HCFA_Info.php

```diff
@@ -0,0 +1,151 @@
+ * @copyright Copyright (c) 2013 Kevin Yeh <kevin.y@integralemr.com>
+ * @copyright Copyright (c) 2018 Stephen Waite <stephen.waite@cmsvt.com>
+ * @link https://github.com/openemr/openemr/tree/master
+ * @license https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Billing;
+
+use OpenEMR\Billing\HCFA_1500;
+
+class HCFA_Info
+{
+    protected $row;
+    protected $column;
+    protected $width;
+    protected $info;
+
+    /**
+     *
+     * @param type $row    Which row to put this data on
+     * @param type $column Which column to put this data in
+     * @param type $width  How many characters max to print on
+     * @param type $info   The text to print on the form at the specified location
+     */
+    public function __construct($row, $column, $width, $info)
+    {
+        $this->row=$row;
+        $this->column=$column;
+        $this->width=$width;
+        $this->info=$info;
+    }
+
+    /**
+     * Determine relative position of an element
+     *
+     * @return type integer
+     */
+    public function get_position()
+    {
+        return $this->row*100+$this->column;
+    }
+
+    /**
+     * Add the info to the form
+     */
+    public function put()
+    {
+        // Override the default value for "strip" with put_hcfa to keep periods
+        HCFA_1500::put_hcfa($this->row, $this->column, $this->width, $this->info, '/#/');
+    }
+
+    /**
+     * comparator function for hfca_info class to allow proper sorting
+     *
+     * @param type $first
+     * @param type $second
+     * @return int
+     */
+    private function cmp_hcfa_info($first, $second)
+    {
+        $first_value = $first->get_position();
+        $second_value = $second->get_position();
+        if ($first_value == $second_value) {
+            return 0;
+        }
+
+        return $first_value < $second_value ? -1 : 1;
+    }
+    /**
+     * calculate where on the form a given diagnosis belongs and add it to the entries
+     *
+     * @param array $hcfa_entries
+     * @param type $number
+     * @param type $diag
+     */
+    function add_diagnosis(&$hcfa_entries, $number, $diag)
+    {
+        /*
+         * The diagnoses go across the page.
+         * Positioned
+         *  A B C D
+         *  E F G H
+         *  I J K L
+         */
+        $column_num = ($number%4);
+        $row_num = (int)($number / 4);
+
+        // First column is at location 3, each column is 13 wide
+        $col_pos=3+13*$column_num;
+
+        // First diagnosis row is 38
+        $strip='/[.#]/';
+        $diag = preg_replace($strip, '', strtoupper($diag));
+        $row_pos=38+$row_num;
+        $hcfa_entries[]=new HCFA_Info($row_pos, $col_pos, 8, $diag);
+    }
+
+    /**
+     * Process the diagnoses for a given claim. log any errors
+     *
+     * @param type $claim
+     * @param string $log
+     */
+    public function process_diagnoses_02_12(&$claim, &$log)
```

Comment:
add a third parameter to take in the $hcfa object

Reply:
ok, disregard above, lets move this function into the HCFA_1500 function as a private function